cardano-node/src/Cardano/Node/Configuration/POM.hs

```diff
@@ -0,0 +1,328 @@
+    }
+
+instance Semigroup PartialNodeConfiguration where
+  (<>) = gmappend
+
+instance Monoid PartialNodeConfiguration where
+  mempty = PartialNodeConfiguration
+             mempty mempty mempty mempty mempty
+             mempty mempty mempty mempty mempty
+             mempty mempty mempty mempty mempty mempty
+  mappend x y =
+    PartialNodeConfiguration
+      { pncNodeAddr = pncNodeAddr x <> pncNodeAddr y
+      , pncConfigFile = pncConfigFile x <> pncConfigFile y
+      , pncTopologyFile = pncTopologyFile x <> pncTopologyFile y
+      , pncDatabaseFile = pncDatabaseFile x <> pncDatabaseFile y
+      , pncProtocolFiles = pncProtocolFiles x <> pncProtocolFiles y
+      , pncValidateDB = pncValidateDB x <> pncValidateDB y
+      , pncShutdownIPC = pncShutdownIPC x <> pncShutdownIPC y
+      , pncShutdownOnSlotSynced = pncShutdownOnSlotSynced x <> pncShutdownOnSlotSynced y
+      , pncProtocolConfig = pncProtocolConfig x <> pncProtocolConfig y
+      , pncSocketPath = pncSocketPath x <> pncSocketPath y
+      , pncMaxConcurrencyBulkSync = pncMaxConcurrencyBulkSync x <> pncMaxConcurrencyBulkSync y
+      , pncMaxConcurrencyDeadline = pncMaxConcurrencyDeadline x <> pncMaxConcurrencyDeadline y
+      , pncViewMode = pncViewMode x <> pncViewMode y
+      , pncLoggingSwitch = pncLoggingSwitch x <> pncLoggingSwitch y
+      , pncLogMetrics = pncLogMetrics x <> pncLogMetrics y
+      , pncTraceConfig = pncTraceConfig x <> pncTraceConfig y
+      }
+
+
+instance FromJSON PartialNodeConfiguration where
+  parseJSON =
+    withObject "PartialNodeConfiguration" $ \v -> do
+
+      -- Node parameters, not protocol-specific
+      pncSocketPath' <- Last <$> v .:? "SocketPath"
+
+      -- Blockfetch parameters
+      pncMaxConcurrencyBulkSync' <- Last <$> v .:? "MaxConcurrencyBulkSync"
+      pncMaxConcurrencyDeadline' <- Last <$> v .:? "MaxConcurrencyDeadline"
+
+      -- Logging parameters
+      pncViewMode'      <- Last <$> v .:? "ViewMode"
+      pncLoggingSwitch' <- v .:? "TurnOnLogging" .!= True
+      pncLogMetrics'    <- Last <$> v .:? "TurnOnLogMetrics"
+      pncTraceConfig'   <- if pncLoggingSwitch'
+                           then Last . Just <$> traceConfigParser v
+                           else return . Last $ Just TracingOff
+
+      -- Protocol parameters
+      protocol <-  v .:? "Protocol" .!= ByronProtocol
+      pncProtocolConfig' <-
+        case protocol of
+          ByronProtocol ->
+            Last . Just . NodeProtocolConfigurationByron <$> parseByronProtocol v
+
+          ShelleyProtocol ->
+            Last . Just . NodeProtocolConfigurationShelley <$> parseShelleyProtocol v
+
+          CardanoProtocol ->
+            Last . Just  <$> (NodeProtocolConfigurationCardano <$> parseByronProtocol v
+                                                               <*> parseShelleyProtocol v
+                                                               <*> parseHardForkProtocol v)
+      pure mempty {
```

Comment:
Why `mempty` there? Would it not be better to use the actual struct name?

Reply:
I agree, it makes it clearer what I'm doing.